apis/appconfiguration/v1alpha1/keyvalue_types.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+)
+
+// KeyValueParameters are the configurable fields of a KeyValue.
+type KeyValueParameters struct {
+	// Endpoint of the App Configuration.
+	Endpoint string `json:"endpoint"`
+
+	// Key of the KeyValue.
+	// +immutable
+	Key string `json:"key"`
+
+	// Label of the KeyValue.
+	// +immutable
+	Label string `json:"label"`
+
+	// Value of the KeyValue
+	Value string `json:"value"`
+
+	// Content-Type of the KeyValue.
+	// +optional
+	ContentType *string `json:"contentType,omitempty"`
+
+	// Whether this KeyValue is locked.
+	// +optional
+	Locked *bool `json:"locked,omitempty"`
+
+	// Tags of the KeyValue resource.
+	// +optional
+	Tags *map[string]string `json:"tags,omitempty"`
+}
+
+// A KeyValueSpec defines the desired state of a KeyValue.
+type KeyValueSpec struct {
+	xpv1.ResourceSpec `json:",inline"`
+	ForProvider       KeyValueParameters `json:"forProvider"`
+}
+
+// KeyValueObservation represents the observed state of the Secret object in Azure.
+type KeyValueObservation struct {
+	// Time that the KeyValue was last modified.
```

Comment:
Looking the controller test. I see that you are using the date format. Is this field `time` or `date` format? I think that this format should be in documentation